extensions/spring-data-jpa/deployment/src/main/java/io/quarkus/spring/data/deployment/MethodNameParser.java

```diff
@@ -516,10 +599,13 @@ private FieldInfo getField(String fieldName) {
                 superClassType = parameterizedType.owner();
             }
         }
-        if (mappedSuperClassInfos.size() > 0) {
-            return mappedSuperClassInfos;
-        }
-        return Collections.emptyList();
+        return mappedSuperClassInfoElements;
+    }
+
+    private static List<String> splitCamelCaseString(String s) {
+        List<String> result = new ArrayList<>();
+        Collections.addAll(result, s.split("(?<!(^|[A-Z]))(?=[A-Z])|(?<!^)(?=[A-Z][a-z])"));
```

Comment:
Perhaps add comment about how this works? Regex can be daunting when coming back to them after not having seen them for a while :)

Reply:
Do we have a "policy" regarding pre-compiling `Pattern`? Or is ok here if it is re-compiled every time since it probably won't be thousands of invocations and it is "just" deployment code?

https://www.baeldung.com/java-regex-pre-compile

Reply:
We don't really have one, but in this case it's very unlikely we will run into a bottleneck due to the very small number of methods this case handles.

Reply:
@geoand I thought the name of the method was explicit 😅  I will add a comment anyway..
@famod yes it's a code executed only once..

Reply:
It is, I just want an explanation of the regex itself if possible 😎

Reply:
@geoand I found a better choice.. in the apache commons there is the same method `StringUtils.splitByCharacterTypeCamelCase()` 😁

Reply:
Even better!